include/seqan3/range/container/aligned_allocator.hpp

```diff
@@ -28,6 +28,12 @@ namespace seqan3
  *
  * \details
  *
+ * This class allocates memory at the given alignment boundary. If the specified `alignment` is not supported by the
+ * used allocation method a std::bad_alloc exception will be thrown. For requested alignments larger than
+ * `__STDCPP_DEFAULT_NEW_ALIGNMENT__`, also called new-extended alignments, the storage will have the alignment
+ * specified by the value `alignment`. Otherwise, the storage is aligned for any object that does not have new-extended
+ * alignment, e.g. int or double, and is of the requested size.
```

Comment:
From this text, I do not understand what a `memory alignment` is.
If you think this is common knowledge, can you give just me an explanation? If not, one or few sentences would be great since this is a public interface.  

Side note: I see that the number is always a multiple of the word size. Is this required and if so do you static_assert this?

Reply:
> Side note: I see that the number is always a multiple of the word size. Is this required and if so do you static_assert this?

I am not sure I understand what do you mean?